src/qibolab/platforms/abstract.py

```diff
@@ -134,24 +137,47 @@ def reload_settings(self):
         self.relaxation_time = settings["settings"]["repetition_duration"]
         self.sampling_rate = settings["settings"]["sampling_rate"]
 
-        # TODO: Create better data structures for native gates
+        # Load native gates
         self.native_gates = settings["native_gates"]
-        self.native_single_qubit_gates = self.native_gates["single_qubit"]
-        if "two_qubit" in self.native_gates:
-            self.native_two_qubit_gates = self.native_gates["two_qubit"]
-            for gates in self.native_gates["two_qubit"].values():
-                self.two_qubit_natives |= set(gates.keys())
-        else:
-            # dummy value to avoid transpiler failure for single qubit devices
-            self.two_qubit_natives = {"CZ"}
-
         # Load characterization settings and create ``Qubit`` and ``Channel`` objects
         for q in settings["qubits"]:
             if q in self.qubits:
                 for name, value in settings["characterization"]["single_qubit"][q].items():
                     setattr(self.qubits[q], name, value)
             else:
-                self.qubits[q] = Qubit(q, **settings["characterization"]["single_qubit"][q])
+                self.qubits[q] = qubit = Qubit(q, **settings["characterization"]["single_qubit"][q])
+                # register channels to qubits when we are using the old format
+                # needed for ``NativeGates`` to work
+                if "qubit_channel_map" in self.settings:
+                    ro, qd, qf, _ = self.settings["qubit_channel_map"][q]
+                    if ro is not None:
+                        qubit.readout = Channel(ro)
+                    if qd is not None:
+                        qubit.drive = Channel(qd)
+                    if qf is not None:
+                        qubit.flux = Channel(qf)
+                # register single qubit native gates to Qubit objects
+                qubit.native_gates = NativeSingleQubitGates.from_dict(qubit, self.native_gates["single_qubit"][q])
+        if "two_qubit" in self.native_gates:
+            for pair, gatedict in self.native_gates["two_qubit"].items():
+                qubit1, qubit2 = pair.split("-")
```

Comment:
The problem here is that `self.native_gates["two_qubit"]` is a dictionary loaded from the runcard yaml and I believe yaml does not support tuples.

Reply:
> Then you can use a 2-lists, that is more reasonable for YAML. Python is able to unpack many things.
> 
> ```python
> a, b = [0, 1]
> assert a == 0 and b == 1
> ```

I am not sure if that's possible in our case, because after loading the YAML the 2-list would be a key of a dictionary which cannot happen since lists are mutable. However I did not check if it is possible to write a YAML with list as key and then load it in Python.

> Btw: YAML support arbitrary objects, worst case saving them as binary blobs. For standard library object might even be reasonable to use them, since the standard library is almost always available. But if we restrict to YAML native types, we could load also in a different language and exotic contexts, so I'd avoid serializing tuples.

Also, I believe serializing would be against our main motivation of using YAML which is so that the runcard is readable (and manually editable...).

Reply:
> I am not sure if that's possible in our case, because after loading the YAML the 2-list would be a key of a dictionary which cannot happen since lists are mutable. However I did not check if it is possible to write a YAML with list as key and then load it in Python.

Indeed I am getting an error
```
yaml.constructor.ConstructorError: while constructing a mapping
found unhashable key
```